addons/project/models/project.py

```diff
@@ -520,6 +526,11 @@ def write(self, vals):
                 project.message_subscribe(project.partner_id.ids)
         if vals.get('privacy_visibility'):
             self._change_privacy_visibility()
+        if 'stage_id' in vals:
+            if self.stage_id.sms_template_id and self.partner_id:
+                self._message_sms_with_template(
```

Comment:
This method needs a singleton only, so if the self contains more than one record, the method will raise an error.